### ISSUE TITLE
Angular 5 support

### ISSUE THREAD

**Issue by siloor** [lentschi/ngx-ellipsis#1]
Hello @lentschi !

At first, thank you for this great package, it worked like charm by our project!

Now we are trying to upgrade to angular 5, and sadly the package throws the following error by the building of the application with aot:

`
ERROR in Error: Error encountered resolving symbol values statically. Function calls are not supported. Consider replacing the function or lambda with a reference to an exported function (position 194:50 in the original .ts file), resolving symbol NgModule in /home/user/Projects/app-ellipsis/ngx-ellipsis/node_modules/@angular/core/core.d.ts, resolving symbol EllipsisModule in /home/use/Projects/app-ellipsis/ngx-ellipsis/dist/index.d.ts, resolving symbol EllipsisModule in /home/user/Projects/app-ellipsis/ngx-ellipsis/dist/index.d.ts
`

Do you plan to upgrade the lib to support Angular 5?

**Comment by nirali-ga** [comment]
Hello, I am using this on my Angular 4 project, but don't know well how to apply as mentioned in Doc  and also it is not showing any result after implementing on screen,any help would be appritiated.I have added following in my html page after importing it in main module.


`  <div ellipsis>Loren ipsum 123 Loren ipsum 123 Loren ipsum 123 Loren ipsum 123 Loren ipsum 123Loren ipsum 123 Loren 23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum23 Loren ipsum 123Loren ipsum </div>`

**Comment by lentschi** [comment]
@siloor Thanks for reporting this! I can confirm this issue and will try to look into it as soon as I can find some time.
What I found out so far: It is not a problem with the directive itself: If you copy it to your angular5 project manually, it works as advertised. So there must be some problem with the build process, but I couldn't find out what yet. (Any help appreciated!)

**Comment by lentschi** [comment]
@nirali-ga Hmm - ngx-ellipsis doesn't come with any styles. The ellipsis will only apply, if the div you're using is confined by something or has its own limited width/height. Have you tried adding `style="width:100px; height:100px"` to your div?
If this is not the cause for your trouble, please create a new issue with more details (exact angular version etc., or possibly even a plunkr) as I do not believe this is in any way related to siloor's issue...

**Comment by lentschi** [comment]
I finally figured it out - All I had to do was upgrade node and npm before running `npm run build`. Version 1.0.6 seems to work with angular 4 _and_ 5.

**Comment by siloor** [comment]
Thanks for the prompt fix @lentschi , it did really solved the building issue!

Although one issue remained. Now it says by running the script:
```
module.js:471
    throw err;
    ^

Error: Cannot find module './lib/directives/ellipsis.directive'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at /home/mmagyar/Projects/foodtodo/node_modules/ngx-ellipsis/bundles/ngxEllipsis.umd.js:10:28
    at Object.defineProperty.value (/home/mmagyar/Projects/foodtodo/node_modules/ngx-ellipsis/bundles/ngxEllipsis.umd.js:2:65)
    at Object.<anonymous> (/home/mmagyar/Projects/foodtodo/node_modules/ngx-ellipsis/bundles/ngxEllipsis.umd.js:5:2)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```

The problem is, that in the `bundles/ngxEllipsis.umd.js` this line
`var ellipsis_directive_1 = require("./lib/directives/ellipsis.directive");`
should be this:
`var ellipsis_directive_1 = require("../lib/directives/ellipsis.directive");`

After that change everything works properly.

Could you please make this change on the code base?
I just don't want to do it, because maybe you would want to move the `bundles/ngxEllipsis.umd.js` to the root, so the path would be good like this.

**Comment by lentschi** [comment]
@siloor Do you have a hint on _how_ I could do that? ;-) (`bundles/ngxEllipsis.umd.js` is a file generated by `npm run build` - Of course I could edit it manually before publishing, but that process would be very error-prone)
Also: How are you using it? I don't get the error and I tried it on clean installs of angular 4 and 5 when executing `ng new project && cd project && npm install ngx-ellipsis && ng serve` - So why are you getting the error and I am not? (I don't want to break the package for others...)

**Comment by siloor** [comment]
@lentschi Well, if you would set the destination of the `ngEllipsis.umd.js` to `/dist` and not to `/dist/bundles`, I think that would solve the issue (although I'm not sure that this will be a good design decision).

Do you add the ellipsis module in the app module before running `ng serve`?

**Comment by lentschi** [comment]
@siloor 
> Do you add the ellipsis module in the app module before running ng serve?

Yes, sure I did ;-) I've just uploaded a very simplistic demo project including version 1.0.6 of `ngx-ellipsis`:
https://github.com/lentschi/ngx-ellipsis-angular5-demo
-> After running `npm install` both `npm run start` and `npm run build` work as expected. (If they do not, could you please update your local node and npm to the most recent stable releases?)

So for you to get this error, I assume your build process is somehow different from the default... (Are you using gulp, ionic,...?) `package.json` has a reference to `bundles/ngxEllipsis.umd.js` so by "normal" bundlers it should be found.

However it _also_ works when changing the paths as you suggested, [which I did in the branch issue001](https://github.com/lentschi/ngx-ellipsis/commit/3e2c8bef6189cbb28d084b5bcf208f316b7d1139) (but not yet at npmjs.com). So I consider changing it at npmjs.com as well, if you can tell me what kind of build process it currently impedes in your project...

**Comment by siloor** [comment]
@lentschi Thanks for helping me by this issue!

Sadly I can't show you the project I'm working on (company stuff), but tomorrow I will prepare a simple project where we can examine the error (it is a minor one, so the fix will be something simple for sure).

Anyway, I think in the `ngxEllipsis.umd.js` the `./lib/directives/ellipsis.directive` relative path is bad for sure, so I think this should be a problem.

I will let you know, when I'm ready with the demo project!

**Comment by lentschi** [comment]
@siloor 

> Anyway, I think in the ngxEllipsis.umd.js the ./lib/directives/ellipsis.directive relative path is bad for sure, so I think this should be a problem.

But - as I said - `ngxEllipsis.umd.js` is a _generated_ file. It is created during the build process by [https://rollupjs.org](https://rollupjs.org). None of the [rollup configuration options](https://rollupjs.org/#big-list-of-options) suggest, that it's possible not to use paths starting with `./` (Or what do you mean by "the relative path is bad for sure"?  They could hardly use an _absolute_ path after all ;-) )

The [tutorial I used to bundle the library](https://medium.com/@cyrilletuzi/how-to-build-and-publish-an-angular-module-7ad19c0b4464) is not the most current. So I might try using a library starter kit [such as this](https://github.com/robisim74/angular-library-starter/blob/master/rollup.config.js) instead.

However, could you please first build and try out, if the changes in issue001 already fix it for you...?:

```bash
cd /tmp/
git clone git@github.com:lentschi/ngx-ellipsis.git
cd ngx-ellipsis
git checkout issue001
npm install
npm run build
cd dist
npm pack
cd /path/to/your/project
npm remove --save ngx-ellipsis
npm install --save /tmp/ngx-ellipsis/dist/ngx-ellipsis-1.0.7.tgz
```
 -> Now check, if you still see the `Cannot find module` error.
Btw.: What I find weird about that error of yours, is that `./lib/directives/ellipsis.directive` really _is_ not a module - it's a directive. You aren't by chance trying to import the directive directly...? If so, that is considered bad practice - You need to import the whole `EllipsisModule` (see `Installation` in the README).

**Comment by siloor** [comment]
@lentschi 
I just tried to create a demo application, where we could examine the error. But as it turned out, it only has the issue with server side rendering.

And actually, I found another workaround which works with angular universal as well. I'm not letting the `package.json` to tell which file needs to be imported (`ngxEllipsis.umd.js`), but specify directly the `index.d.ts` using this import line `import { EllipsisModule } from 'ngx-ellipsis/index';`.

Like this everything works properly on both sides (client-server).

So actually the version `v1.0.6` works perfectly with angular 5, no changes needed. Thanks @lentschi for your help! And again, very great lib!
